### PR TITLE
feat: Add support for Python 3.10

### PR DESCRIPTION
**Pull request by mctavish**
This PR adds support for Python 3.10 to the debug agent.  A summary of the key changes are:

* Add handling for new opcodes
* Add support for PEP-626 - modified API for line numbers:
    * Abstract handling of line numbers so that both old and new APIs can be supported
    * Use co_lines() in Python code for enumerating line numbers in a code object
    * Copy in and use the _PyLineTable functions for handling co_linetable in C++ code
    * Modify InsertAndUpdateLineData to update the code object's linetable data using the 3.10 format.  Note that this will need to be modified again for Python 3.11 which uses a condensed version of the linetable.
    * Update tests to handle the linetable changes
* Update tests to handle modified JUMP offsets
* Update tests to handle modified error messages

### CODE REVIEW

File: src/googleclouddebugger/module_explorer.py [hunk 1]
```diff
@@ -78,15 +78,24 @@ def _GetLineNumbers(code_object):
...
-  current_line = code_object.co_firstlineno
-  for line_incr in line_incrs:
-    current_line += line_incr
-    yield current_line
+
+  if sys.version_info.minor < 10:
+    # Get the line number deltas, which are the odd number entries, from the
+    # lnotab. See
+    # https://svn.python.org/projects/python/branches/pep-0384/Objects/lnotab_notes.txt
+    # In Python 3, this is just a byte array.
```
Comment (jasonborg) [inline src/googleclouddebugger/module_explorer.py:86]: Python 3, prior to 3.10

File: src/googleclouddebugger/immutability_tracer.cc [hunk 2]
```diff
@@ -468,6 +477,11 @@ static OpcodeMutableStatus IsOpcodeMutable(const uint8_t opcode) {
     case RERAISE:
     case WITH_EXCEPT_START:
     case LOAD_ASSERTION_ERROR:
+#endif
+#if PY_VERSION_HEX >= 0x030A0000
+    // Added in Python 3.10
+    case COPY_DICT_WITHOUT_KEYS:
```
Comment (jasonborg) [inline src/googleclouddebugger/immutability_tracer.cc:483]: This one may be safe to mark immutable?

**Review by jasonborg — APPROVED** [review]
Looks good! Glad to see this was able to get done, congrats.

File: src/googleclouddebugger/bytecode_manipulator.cc [hunk 3]
```diff
@@ -296,7 +296,7 @@ struct Insertion {
 // InsertAndUpdateBranchInstructions.
 static const int kMaxInsertionIterations = 10;
 
-
+#if PY_VERSION_HEX < 0x030A0000
 // Updates the line number table for an insertion in the bytecode.
 // This is different than what the Python 2 version of InsertMethodCall() does.
```
[the remark this answers is not in the document]
Comment (mctavish) [reply src/googleclouddebugger/bytecode_manipulator.cc:301]: Good point.  I've stripped it down to only what is relevant now.  Much shorter now that it isn't comparing the implementation to a flawed one that no longer exists.

File: src/googleclouddebugger/module_explorer.py (hunk 1 above)
Comment (mctavish) [reply src/googleclouddebugger/module_explorer.py:86]: Updated.

File: src/googleclouddebugger/immutability_tracer.cc (hunk 2 above)
Comment (mctavish) [reply src/googleclouddebugger/immutability_tracer.cc:483]: Agreed.  Moved.

File: src/googleclouddebugger/bytecode_manipulator.cc [hunk 4]
```diff
@@ -330,6 +330,29 @@ static void InsertAndUpdateLnotab(int offset, int size,
     }
   }
 }
+#else
+static void InsertAndUpdateLineData(int offset, int size,
```
[the remark this answers is not in the document]
Comment (mctavish) [reply src/googleclouddebugger/bytecode_manipulator.cc:334]: Added something; let me know what you think.

**Pull request merged**